source/uwp/Renderer/lib/XamlHelpers.cpp

```diff
@@ -1,516 +1,338 @@
+                                // hidden while they are retrieved, we shouldn't hide
+                                // the separator
+                                separator.Visibility(winrt::Visibility::Collapsed);
+                            }
+                            else
+                            {
+                                separator.Visibility(winrt::Visibility::Visible);
+                            }
+                        }
+
+                        foundPreviousVisibleElement |= (visibility == winrt::Visibility::Visible);
+                    }
+                }
+            }
         }
-
-        contentControl.Content(textBlock);
     }
 
-    HRESULT HandleStylingAndPadding(_In_ IAdaptiveContainerBase* adaptiveContainer,
-                                    _In_ IBorder* containerBorder,
-                                    _In_ IAdaptiveRenderContext* renderContext,
-                                    _In_ IAdaptiveRenderArgs* renderArgs,
-                                    _Out_ ABI::AdaptiveCards::ObjectModel::Uwp::ContainerStyle* containerStyle)
+    winrt::ContainerStyle HandleStylingAndPadding(winrt::IAdaptiveContainerBase const& adaptiveContainer,
+                                                  winrt::Border const& containerBorder,
+                                                  winrt::AdaptiveRenderContext const& renderContext,
+                                                  winrt::AdaptiveRenderArgs renderArgs)
     {
-        ABI::AdaptiveCards::ObjectModel::Uwp::ContainerStyle localContainerStyle;
-        RETURN_IF_FAILED(adaptiveContainer->get_Style(&localContainerStyle));
+        winrt::UIElement elem{nullptr};
 
-        ABI::AdaptiveCards::ObjectModel::Uwp::ContainerStyle parentContainerStyle;
-        RETURN_IF_FAILED(renderArgs->get_ContainerStyle(&parentContainerStyle));
+        auto localContainerStyle = adaptiveContainer.Style();
+        auto parentContainerStyle = renderArgs.ContainerStyle();
 
         bool hasExplicitContainerStyle{true};
-        if (localContainerStyle == ABI::AdaptiveCards::ObjectModel::Uwp::ContainerStyle::None)
+        if (localContainerStyle == winrt::ContainerStyle::None)
         {
             hasExplicitContainerStyle = false;
             localContainerStyle = parentContainerStyle;
         }
 
-        ComPtr<IAdaptiveHostConfig> hostConfig;
-        RETURN_IF_FAILED(renderContext->get_HostConfig(&hostConfig));
+        auto hostConfig = renderContext.HostConfig();
 
-        ComPtr<IAdaptiveSpacingConfig> spacingConfig;
-        RETURN_IF_FAILED(hostConfig->get_Spacing(&spacingConfig));
+        auto spacingConfig = hostConfig.Spacing();
 
-        UINT32 padding;
-        RETURN_IF_FAILED(spacingConfig->get_Padding(&padding));
-        DOUBLE paddingAsDouble = static_cast<DOUBLE>(padding);
+        uint32_t padding = spacingConfig.Padding();
+        double paddingAsDouble = (double)padding;
```

Comment:
minor: `(double)` is valid but it's an old c style. 
possible the better is `double paddingAsDouble {padding}`

http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es49-if-you-must-use-a-cast-use-a-named-cast

Reply:
**conversion from 'uint32_t' to 'const double' requires a narrowing conversion** we'll get the warning :)

Reply:
![image](https://user-images.githubusercontent.com/21356912/151237436-5e66f6c0-d740-4a17-a446-04e1f75488d2.png)
more readable and easy to understand:
![image](https://user-images.githubusercontent.com/21356912/151237459-f1e0ab9e-6919-4e2b-9114-74efaae86d9b.png)